Introduction.md is missing CMake headers instructions

vitorroriz opened the pull request:
#### db06fe04ae7477639c88d6195dfe51c094f20d39
<pre>
Introduction.md is missing CMake headers instructions
<a href="https://bugs.webkit.org/show_bug.cgi?id=249229">https://bugs.webkit.org/show_bug.cgi?id=249229</a>
rdar://problem/103303954

Reviewed by Simon Fraser.

* Introduction.md:
Adding instructions about cmake headers.

Canonical link: <a href="https://commits.webkit.org/258045@main">https://commits.webkit.org/258045@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/60f202f7e4feee848372e2a7741f4babc3458499

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/100752 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/9896 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/33793 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/110051 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/10835 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/480 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/93152 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/107897 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/106534 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/8182 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/91431 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/34797 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/90091 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/22824 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/77769 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/3590 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/24348 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/3614 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/474 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/9725 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/43847 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/5391 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2888 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Introduction.md:
```diff
@@ -381,9 +381,11 @@ To do that, turn on the target membership in respective framework as set the mem
 This will ensure the relevant header file is exported from WTF / WebCore to other downstream projects like WebKitLegacy.
 ![Screenshot of exporting a header file](resources/xcode-export-header.png)
 
+WPE and GTK use CMake for building the project. If the header is not platform-specific, you might want to add an entry for it into
```
Can you make the "might" more specific? Is this only necessary if e.g. a WebCore header is included by WebKit?

donny-dont replied:
It is only necessary if its included by WebKit.

Also it isn't just WPE and GTK using CMake. Every non-Cocoa port uses it. I'd say non-Apple but AppleWin uses CMake.

vitorroriz replied:
@donny-dont  How about WebKit/Headers.cmake ?

vitorroriz committed:
Introduction.md is missing CMake headers instructions
https://bugs.webkit.org/show_bug.cgi?id=249229
rdar://problem/103303954

Reviewed by Simon Fraser.

* Introduction.md:
Adding instructions about cmake headers.

Canonical link: https://commits.webkit.org/258045@main
